### DISCUSSION TITLE
How to update Zsh through a shellscript ?

### DISCUSSION THREAD

**Discussion by kevinadhiguna** [ohmyzsh/ohmyzsh#10442] [Help] [answered]
Hi, I am using Zsh and am trying to update Zsh through a shellscript.

This is how it looks : 
```shell
# Update OS
sudo apt update -y
apt list --upgradable
sudo apt upgrade -y
sudo apt autoremove -y
sudo apt autoclean

# Update Zsh
omz update
```

However, this does not update Zsh as it throws an error saying : 
`./update.sh: 7: omz: not found`.

`which omz` showed : 
![image](https://user-images.githubusercontent.com/43397636/142962189-e79765fa-4b1e-48e5-863e-855b7fc8f372.png)

Zsh version : zsh 5.8 (x86_64-ubuntu-linux-gnu)
OS : Ubuntu 20.04

**Comment by mcornella** [comment, marked as answer]
You have to call the `$ZSH/tools/upgrade.sh` script directly. The `omz` command is only available on an interactive zsh session.

**Reply by kevinadhiguna** [reply]
Where is `tools/upgrade.sh` commonly located ?

**Reply by mcornella** [reply]
In the directory where you installed soon Oh My Zsh, which is referenced by the `$ZSH` variable.

**Comment by robbyrussell** [comment]
As Marc was saying:

`$ZSH/tools/upgrade.sh` 

(most commonly in `~/.oh-my-zsh/tools/upgrade.sh`)
